src/lib/query/iface.rs

```diff
@@ -314,6 +317,7 @@ pub(crate) fn parse_nl_msg_to_iface(
     if name.is_empty() {
         return Ok(None);
     }
+    let driver = _get_iface_driver(name.clone());
```

Comment:
Normally, we use `&str` for function unless ownership need to be passing around.

Reply:
Sure, fixed signature to `&str` and function call to be `name.as_str()`